library/api/src/main/java/com/epmedu/animeal/api/storage/StorageApiImpl.kt

```diff
@@ -0,0 +1,36 @@
+package com.epmedu.animeal.api.storage
+
+import android.net.Uri
+import androidx.core.net.toFile
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.storage.options.StorageGetUrlOptions
+import com.epmedu.animeal.common.data.wrapper.ApiResult
+import com.epmedu.animeal.extensions.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+
+internal class StorageApiImpl : StorageApi {
+
+    override suspend fun uploadFile(fileName: String, uri: Uri): ApiResult<Unit> {
+        return suspendCancellableCoroutine {
+            val file = uri.toFile()
+
+            Amplify.Storage.uploadFile(
```

Comment:
I'm almost sure that the Amplify switch from Main Thread under the hood, but unfortunately, I didn't find any proof shortly. I was wondering, is it my suggestion is true.